### ISSUE TITLE
ORTB error responses

### ISSUE THREAD

**Issue by bretg** [prebid/prebid-server#3556]
There's been a request to redesign how PBS responds to errors.

Currently when PBS responds with an HTTP error code (e.g. 400), the body of the response is just a string.

This issue tracks discussion for returning ORTB JSON responses with a No-Bid-Response code when PBS returns an error. e.g.

```
{nbr: x, seatbid [], ext.prebid.errors:{...} }
```

This could be a breaking change, so would have to happen in a major release, and/or be combined with a client request flag defining how they want the server to respond to errors. e.g. `ext.prebid.ortberrors:true`.

Another topic for discussion is whether to combine this with a more general review and refactoring of error codes in general.

**Comment by SyntaxNode** [comment]
We should discuss how Prebid Server responds in the case of a request with invalid json.

PBS-Go will return an error which includes details of the invalid json. There's a request from @maditya to return an error message without details because the current behavior violates best practices of not directly returning a framework error message which has security implications.

**Comment by pm-nikhil-vaidya** [comment]
PRD

https://docs.google.com/document/d/1JiLWHHpsq9ou-pV7nPk9UVHl6hkj2ke2tTct2tcxiOU

**Comment by bsardo** [comment]
@pm-nikhil-vaidya is PubMatic still interested in this feature? I don't see any recent activity on the PRD.

**Comment by pm-nikhil-vaidya** [comment]
@bsardo 
Yes we are working on it, developed initial version of this. Will contribute the changes soon.

**Comment by bsardo** [comment]
@pm-nikhil-vaidya just checking in if you have a timeframe on when you expect to deliver this.

**Comment by bsardo** [comment]
This could be a 4.0 item. @Net-burst and I will discuss offline.

**Comment by bsardo** [comment]
@Net-burst did Java add this as part of 4.0? Go did not. I'm wondering if anyone in the community is interested in this feature making it worth the effort for 5.0.

**Comment by Net-burst** [comment]
@bsardo nope, Java also didn't include it in 5.0. If I remember correctly, we agreed to make a toggle for it, so it's not a breaking change for all intents and purposes too.
